db/migrate/20200803113903_migrate_host_type_in_host_config_groups.foreman_puppet.rb

```diff
@@ -51,4 +33,38 @@ def down
                          .update_all(host_type: 'Hostgroup', host_id: hostgroup_facet_ids[hostgroup_facet_id])
     end
   end
+
+  private
+
+  def migrate_host_data_to_facets
+    host_config_group_ids = FakeHostConfigGroup.for_host.pluck(:host_id).uniq
+    host_config_group_ids.each do |host_id|
+      host_facet_id = ForemanPuppet::HostPuppetFacet.where(host_id: host_id).pick(:id)
+      host_groups = FakeHostConfigGroup.for_host.where(host_id: host_id)
+      if host_facet_id
+        host_groups.update_all(host_type: 'ForemanPuppet::HostPuppetFacet', host_id: host_facet_id)
+      else
+        deleted_groups = FakeConfigGroup.where(id: host_groups.pluck(:config_group_id)).pluck(:name)
+        say "deleting groups #{deleted_groups.join(', ')} from Host (id=#{host_id}) because it doesn't exists or has no environment"
+        host_groups.delete_all
```

Comment:
What if a config group belongs to multiple hosts of which one was deleted? Then it should remain. Does this hold true?

Reply:
Yes, this will run only the SQL, so it will delete nothing else then the joining rows from the join table. Config group will be always kept.